sqlmesh/core/dialect.py

```diff
@@ -89,6 +90,10 @@ def output_name(self) -> str:
         return self.this.name
 
 
+class StagedFilePath(exp.Table):
+    """Represents paths to "staged files" in Snowflake."""
+
```

Comment:
I decided not to pollute SQLGlot with this definition on purpose: the issue caused by `@` is SQLMesh-specific, so no need to subtype `Table` and introduce unnecessary complexity upstream.